### PR TITLE
isless

### PR DESCRIPTION
**Pull request by shashi**
Implements `Base.isless` on NamedTuples.

### CODE REVIEW

**Comment by davidanthoff** [comment]
This seems too broad to me, I think I would prefer a function signature like this:
````julia
function Base.isless{T<:NamedTuple}(t1::T, t2::T)
````
I.e. if one can only compare named tuples that have the same fields (both in terms of names and types).

We also don't define ``isless`` for two distinct ``immutable`` types, and that seems the relevant comparison to me.

Also, this should probably be a generated function that unrolls the loop so that things are type stable?

**Comment by shashi** [comment]
my main motivation for not using diagonal dispatch was to allow

```julia
julia> @NT(x=1,y=2) < @NT(x=1, y=2, z=1)
true
```

which is analogous to
```julia
julia> (1,2) < (1,2,3)
true
```

I agree the current situation is too broad. I'll have to check if diagonal dispatch is fine for my purposes.

**Comment by shashi** [comment]
One more problem with `Base.isless{T<:NamedTuple}(t1::T, t2::T)`: we can't compare Int fields with Float fields - which seems elementary.

I wonder if there's a way to match only the `_NT_x_y` part of the type without the parameters with the new type system. I can't think of a way off the top of my head.

**Comment by shashi** [comment]
updated to use a generated function to do this.

File: src/NamedTuples.jl [hunk 1]
```diff
@@ -280,6 +280,22 @@ end
     end
 end
 
+@generated function Base.isless(t1::NamedTuple, t2::NamedTuple)
+    if !isequal(fieldnames(t1), fieldnames(t2))
+        throw(ArgumentError("NamedTuple inputs to isless must have the same fields in the same order"))
+    end
+    quote
+        for i = 1:nfields(t1)
+            a = getfield(t1, i)
```
Comment (davidanthoff) [inline src/NamedTuples.jl]: ``a`` will be a type-instability, right? Couldn't you unroll the loop in the generated function and that would give you a type-stable implementation?

Comment (shashi) [inline src/NamedTuples.jl]: fair enough, will update.

**Comment by shashi** [comment]
Done. Ready to merge!

**Comment by shashi** [comment]
ping @mdcfrancis

**Pull request merged**